docs/proposals/images/machinepool-machines/inframachinepool-create-replicas.plantuml

```diff
@@ -0,0 +1,17 @@
+@startuml inframachinepool-create-replicas
+:InfraMachinePool create replicas reconciliation;
+if (status.InfrastructureMachineKind is set on InfraMachinePool) then (no)
+  :set status.InfrastructureMachineKind to InfraMachinePoolMachine;
+else (yes)
+endif
+repeat 
+  :create provider specific resource representing a replica;
+repeat while (~# replicas < # desired replicas ?) is (yes)
+->no;
+repeat 
+  :create InfraMachinePoolMachine representing a replica;
+repeat while (every replica has an InfraMachinePoolMachine?) is (no)
```

Comment:
same thing with this, we should first check if every replica has an InfraMachinePoolMachine before doing "create InfraMachinePoolMachine representing a replica"